stripe/src/resources/charge.rs

```diff
@@ -4,21 +4,119 @@ use params::{List, Metadata, RangeQuery, Timestamp};
 use resources::{Address, Currency, PaymentSource, PaymentSourceParams, Refund};
 use serde_qs as qs;
 
+/// The resource representing a Stripe charge object outcome.
+///
+/// For more details see [https://stripe.com/docs/api#charge_object-outcome](https://stripe.com/docs/api#charge_object-outcome)
 #[derive(Debug, Deserialize, Serialize)]
 pub struct ChargeOutcome {
     #[serde(rename = "type")]
-    pub outcome_type: String, // (authorized, manual_review, issuer_declined, blocked, invalid)
-    pub network_status: String, // (approved_by_network, declined_by_network, not_sent_to_network, reversed_after_approval)
+    pub outcome_type: OutcomeType,
+    pub network_status: NetworkStatus,
     #[serde(default)]
-    pub reason: Option<String>,
+    pub reason: Option<OutcomeReason>,
     #[serde(default)]
-    pub risk_level: Option<String>, // (normal, elevated, highest, not_assessed, unknown)
+    pub risk_level: Option<RiskLevel>,
     #[serde(default)]
     pub seller_message: Option<String>,
     #[serde(default)]
     pub rule: Option<String>,
 }
 
+/// The resource representing a Stripe charge object outcome outcome_type.
+///
+/// For more details see [https://stripe.com/docs/api#charge_object-outcome-type](https://stripe.com/docs/api#charge_object-outcome-type)
+#[derive(Deserialize, Serialize, PartialEq, Debug, Clone, Eq)]
+#[serde(rename_all = "snake_case")]
+pub enum OutcomeType {
+    Authorized,
+    ManualReview,
+    IssuerDeclined,
+    Blocked,
+    Invalid,
+    #[serde(other)]
+    Other,
+}
+
+/// The resource representing a Stripe charge object outcome network_status.
+///
+/// For more details see [https://stripe.com/docs/api#charge_object-outcome-network_status](https://stripe.com/docs/api#charge_object-outcome-network_status)
+#[derive(Deserialize, Serialize, PartialEq, Debug, Clone, Eq)]
+#[serde(rename_all = "snake_case")]
+pub enum NetworkStatus {
+    ApprovedByNetwork,
+    DeclinedByNetwork,
+    NotSentToNetwork,
+    ReversedAfterApproval,
+    #[serde(other)]
+    Other,
+}
+
+/// The resource representing a Stripe charge object outcome risk level.
+///
+/// For more details see [https://stripe.com/docs/api#charge_object-outcome-risk_level](https://stripe.com/docs/api#charge_object-outcome-risk_level)
+#[derive(Deserialize, Serialize, PartialEq, Debug, Clone, Eq)]
+#[serde(rename_all = "snake_case")]
+pub enum RiskLevel {
+    Normal,
+    Elevated,
+    Highest,
+    NotAssessed,
+    #[serde(other)]
+    Unknown
+}
+
+/// The resource representing a Stripe charge object outcome reason.
+///
+/// For more details see [https://stripe.com/docs/api#charge_object-outcome-reason](https://stripe.com/docs/api#charge_object-outcome-reason)
+#[derive(Deserialize, Serialize, PartialEq, Debug, Clone, Eq)]
+#[serde(rename_all = "snake_case")]
+pub enum OutcomeReason {
+    ApprovedWithID,
```

Comment:
This variant may not serialize correctly (e.g. `approved_with_i_d`, rather than `approved_with_id`).

I would prefer it be renamed to `ApprovedWithId`, which is consistent both with our `SourceId` and `TokenId` types, but also with a larger part of the rust ecosystem that uses camel cases for abbreviations in names (e.g. `HttpRequest`)